test/framework/tinkerbell.go

```diff
@@ -146,8 +149,11 @@ func (t *Tinkerbell) WithProviderUpgrade(fillers ...api.TinkerbellFiller) Cluste
 	}
 }
 
-func (t *Tinkerbell) CleanupVMs(_ string) error {
-	return nil
+// CleanupMachines runs a clean up the Tinkerbell machines which simply powers them down.
+func (t *Tinkerbell) CleanupMachines(_ string) error {
+	ctx, done := context.WithTimeout(context.Background(), 2*time.Minute)
```

Comment:
This context appears to be shared between all bmclib calls to all machines in each test run. This means that all machines have a combined duration of 2minutes for all `GetPowerState` and `SetPowerState` that run. This might not be enough time. BMC's can be notoriously slow. Also, if any call takes the length of the duration then all subsequent calls will fail as the context will have been canceled/timed out. I recommend adding the timeout in `powerOffHardware`.

Reply:
Ah, thanks. Good catch., I'll move it there as suggested.